python/toolbox/metrics.py

```diff
@@ -105,8 +105,11 @@ def log_sample(this_file_id: str, desc: object, names: object, sample: object):
         # This is how we track which element in the metrics array belongs to this metric type
         metric_idx[label] = len(metric_types)
         idx = metric_idx[label]
-        # store the metric_desc info
-        metric_type = { 'desc': desc, 'names': names }
+        metric_type = { 'desc': {}, 'names': {} }
+        for key in desc.keys():
+            metric_type['desc'][key] = desc[key]
+        for key in names.keys():
+            metric_type['names'][key] = names[key]
```

Comment:
Hmm, take a look at `copy.deepcopy()`.  I use this all over for resolving this issue:

```
import copy
...
foo = copy.deepcopy(bar)
```

There are numerous examples, several are easily spotted in https://github.com/perftool-incubator/multiplex/blob/master/multiplex.py

Reply:
I just tested a shallow copy in my env and that worked so we might not need the extra import and the deep copy.
```
metric_type = { 'desc': None, 'names': None }
metric_type['desc'] = desc.copy()
metric_type['names'] = names.copy()
```

Reply:
I don't know why, but when I tried this exact same thing before I did the loop, I could not get it to work.  Anyway, I agree the deep copy is not necessary.  I suppose I must have had a typo or something in my attempt to use dict.copy and I can try it again.

Reply:
I probably resort to copy.deepcopy() a little too aggressively out of habit...it makes it where I don't have to worry about the complexity of the data structure I'm copying...